pandas/io/common.py

```diff
@@ -232,13 +233,46 @@ def file_path_to_url(path):
 }
 
 
+def _get_compression_method(compression: Optional[Union[str, Dict[str, Any]]]):
+    """
+    Simplifies a compression argument to a compression method string and
+    a dict containing additional arguments.
+
+    Parameters
+    ----------
+    compression : str or dict
+        If string, specifies the compression method. If dict, value at key
+        'method' specifies compression method.
+
+    Returns
+    -------
+    tuple of ({compression method}, any
+              {compression arguments}, dict)
+
+    Raises
+    ------
+    ValueError on dict missing 'method' key
+    """
+    # Handle dict
+    if isinstance(compression, dict):
+        compression_args = compression.copy()
+        try:
+            compression = compression['method']
+            compression_args.pop('method')
+        except KeyError:
+            raise ValueError("If dict, compression "
+                             "must have key 'method'")
+    else:
+        compression_args = {}
+    return compression, compression_args
```

Comment:
Any reason to return compression and compression_args here? The former will always match whatever is being passed to the function anyway so seems like we could move the `isinstance` check outside of this function and only call this method when `compression` is a dict